js/plugin/popup-dialog.ts

```diff
@@ -1,199 +1,168 @@
-import { LitElement, html, css, HTMLTemplateResult, PropertyValues } from "lit";
-import { property, query } from "lit/decorators.js";
+import { LitElement, html, css } from "lit";
+import { property, state } from "lit/decorators.js";
 import { unsafeHTML } from "lit/directives/unsafe-html.js";
 import { repeat } from "lit/directives/repeat.js";
-import {ifDefined} from 'lit/directives/if-defined.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import structuredClone from "@ungap/structured-clone";
+
 import {
   ensureArray,
   provideHass,
   selectTree,
 } from "../helpers";
 import { loadHaForm } from "../helpers";
 import { ObjectSelectorMonitor } from "../object-selector-monitor";
-import { icon } from "./types";
+import { DialogWidth, HaButtonElement, IconProps, PopupStyle } from "../types/types";
 import { BrowserModPopupParams } from "./popups";
-import structuredClone from "@ungap/structured-clone";
-
 export class BrowserModPopup extends LitElement {
-  @property() open;
-  @property() content;
-  @property() title;
-  @property({ reflect: true }) actions;
-  @property({ reflect: true }) card;
-  @property() right_button;
-  @property() right_button_variant;
-  @property() right_button_appearance;
-  @property() right_button_close;
-  @property() left_button;
-  @property() left_button_variant;
-  @property() left_button_appearance;
-  @property() left_button_close;
-  @property() dismissable;
-  @property({ type: Array}) icons: icon[];
-  @property() tag;
-  @property() dismiss_icon;
-  @property({ reflect: true }) timeout_hide_progress;
-  @property() _style;
-  @property() _formDataValid;
-  @property({type: Array}) _styleAttributes: boolean [];
-  @query("ha-dialog") dialog: any;
-  _autoclose;
-  _autocloseListener;
-  _actions;
-  timeout;
-  _timeoutStart;
-  _timeoutTimer;
-  _resolveClosed;
-  _formdata;
-  card_mod;
-  _popupStyles;
-  _objectSelectorMonitor: ObjectSelectorMonitor;
-  _initialStyle: string;
-  _styleSequence: string[];
-  _styleSequenceIndex: number;
-
-  connectedCallback() {
-    super.connectedCallback();
-    this._objectSelectorMonitor = new ObjectSelectorMonitor(
-      this,
-      (value: boolean) => { this._formDataValid = value }
-    );
-    // When reopening popup, keep style attributes
-    // but make sure they are all set to false
-    this._styleAttributes = this._styleAttributes || [];
-    Object.keys(this._styleAttributes).forEach((key) => {
-      this._styleAttributes[key] = false;
-    });
+  @property({ type: Boolean }) open: boolean = false;
+  @property({ type: String }) width: DialogWidth = "medium";
+  @property({ type: Boolean }) preventScrimClose: boolean = false;
+  @property({ type: String }) headerTitle: string = "";
+  @property({ type: String }) headerSubtitle: string = "";
```

Comment:
Shere is headerSubtitle and headerSubtitlePosition used?

Reply:
See above: Preparation for adoption ha-wa-dialog